clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractCoordinator.java

```diff
@@ -175,8 +177,13 @@ protected abstract void onJoinComplete(int generation,
      */
     public void ensureCoordinatorReady() {
         while (coordinatorUnknown()) {
-            RequestFuture<Void> future = sendGroupCoordinatorRequest();
-            client.poll(future);
+            if (findCoordinatorFuture == null)
+                findCoordinatorFuture = sendGroupCoordinatorRequest();
```

Comment:
I think I found a nicer way to do this. Check out the latest commit and see if you agree.